### PR TITLE
added Sum

### PR DESCRIPTION
**Pull request by annzhukovets**

### CODE REVIEW

**Comment by haydnv** [comment]
The bug you found in `Shape.reduce` is fixed as of commit 4e9b6fe

File: client/tinychain/collection/tensor.py [hunk 1]
```diff
@@ -635,6 +647,69 @@ def gradients(self, loss):
...
+        from tinychain.collection.tensor import Dense
+
+        return Dense.ones_like(self.subject)
+
+    def gradients(self, loss):
+        if self.args is None:
+            loss = self.backward() * loss
+        else:
+            axis = self.args
+            if isinstance(self.subject.shape, Shape):
```
Comment (haydnv) [inline client/tinychain/collection/tensor.py:690]: You should just be able to call `self.subject.shape.reduce([axis])`

File: client/tinychain/collection/tensor.py [hunk 2]
```diff
@@ -635,6 +647,69 @@ def gradients(self, loss):
...
+
+# TODO: make it work for multiple functions and for .sum(axis=tuple)
+class Sum(Reduce):
+    def forward(self):
+        return NDArray.sum(self.subject, axis=self.args)
+
+    def backward(self, variable=None):
+        from tinychain.collection.tensor import Dense
+
+        return Dense.ones_like(self.subject)
```
Comment (haydnv) [inline client/tinychain/collection/tensor.py:683]: This will only work in the case that `self.subject` is a `Variable`. I think what you want is `return derivative_of(self.subject, variable).sum(self.args)`.

**Review by haydnv — CHANGES_REQUESTED** [review]
